xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -55,24 +56,64 @@ CPVRClients::CPVRClients(void) :
 
 CPVRClients::~CPVRClients(void)
 {
-  CAddonMgr::GetInstance().UnregisterAddonMgrCallback(ADDON_PVRDLL);
+  CServiceBroker::GetBinaryAddonManager().UnregisterCallback(ADDON_PVRDLL);
   Unload();
 }
 
 void CPVRClients::Start(void)
 {
-  CAddonMgr::GetInstance().RegisterAddonMgrCallback(ADDON_PVRDLL, this);
+  CServiceBroker::GetBinaryAddonManager().RegisterCallback(ADDON_PVRDLL, this);
 
   UpdateAddons();
 }
 
+void CPVRClients::EnableEvent(BinaryAddonBasePtr addon)
+{
+  UpdateAddon(addon, true);
+  CServiceBroker::GetPVRManager().Start();
+}
+
+void CPVRClients::DisableEvent(BinaryAddonBasePtr addon)
+{
+  UpdateAddon(addon, false);
+  CServiceBroker::GetPVRManager().Start();
+}
+
+void CPVRClients::UpdateEvent(BinaryAddonBasePtr addon)
+{
+  CServiceBroker::GetPVRManager().Stop();
+  StopClient(addon, false);
+}
+
+void CPVRClients::InstallEvent(BinaryAddonBasePtr addon, bool update, bool modal)
+{
+  CServiceBroker::GetPVRManager().Stop();
+  UpdateAddons();
+}
+
+void CPVRClients::PreUnInstallEvent(BinaryAddonBasePtr addon)
+{
+  CServiceBroker::GetPVRManager().Stop();
+  StopClient(addon, false);
+}
+
+void CPVRClients::PostUnInstallEvent(BinaryAddonBasePtr addon)
+{
+  UpdateAddons();
+}
+
+void CPVRClients::RequestRestartEvent(BinaryAddonBasePtr addon)
+{
+  StopClient(addon, true);
+}
+
```

Comment:
Thats right, I checked them currently only with pvr.demo and need a test with bigger parts.

I think with the use of "Actor::Protocol" is it more safe.